awx/main/models/inventory.py

```diff
@@ -1582,5 +1583,19 @@ class tower(PluginFileInjector):
     collection = 'awx'
 
 
+class insights(PluginFileInjector):
+    plugin_name = 'insights'
+    namespace = 'redhat'
+    collection = 'insights'
+
+    def get_plugin_env(self, inventory_update, private_data_dir, private_data_files):
+        credential = inventory_update.get_cloud_credential()
+        ret = super(insights, self).get_plugin_env(inventory_update, private_data_dir, private_data_files)
+        if credential:
+            ret['INSIGHTS_USER'] = credential.get_input('username', default='')
+            ret['INSIGHTS_PASSWORD'] = credential.get_input('password', default='')
```

Comment:
We don't need this and it not DRY. See other examples like rhv:

https://github.com/ansible/awx/blob/698844db4736421ef5dff7b684a4d0870cecd9b3/awx/main/models/inventory.py#L1556

That will tell it to fill in the env vars from the credential type definition at:

https://github.com/ansible/awx/blob/698844db4736421ef5dff7b684a4d0870cecd9b3/awx/main/models/credential/__init__.py#L1070

For your insights credential type, you've already put `INSIGHTS_USER` in the `injector`. So add `base_injector = 'template' ` and then remove the entire `get_plugin_env` method here.

If people dislike this strongly, then we should get rid of `base_injector` and _always_ use the managed/template injector logic if it's present for the credential type. Otherwise we repeat ourselves.

Reply:
Updated.